cmd/antrea-agent/options.go

```diff
@@ -105,9 +105,17 @@ func (o *Options) validate(args []string) error {
 			return fmt.Errorf("Mode %s requires AntreaProxy to be enabled", o.config.TrafficEncapMode)
 		}
 		if o.config.EnableIPSecTunnel {
-			return fmt.Errorf("IPSec tunnel may only be enabled on %s mode", config.TrafficEncapModeEncap)
+			return fmt.Errorf("IPsec tunnel may only be enabled on %s mode", config.TrafficEncapModeEncap)
 		}
 	}
+	if o.config.NoSNAT && encapMode != config.TrafficEncapModeNoEncap {
+		return fmt.Errorf("NoSNAT is applicable to only %s mode", config.TrafficEncapModeEncap)
+	}
```

Comment:
does that mean there will be an error if I explicitly set `noSNAT` to `true` in the config, along with `networkPolicyOnlyMode`. Isn't that confusing given the code below that automatically sets `noSNAT` to true for `networkPolicyOnlyMode`?

Reply:
My intention is: for user populated options, noSNAT can be set only for noEncap, not for others inc. policyOnly. Feel this is simpler for users to understand than requiring noSNAT set to true for policyOnly.
But in agent code, I reused the same option to indicate the behavior of no SNAT and automatically set it for policyOnly. If you think this is confusing, I can modify the code not to auto set noSNAT for the policyOnly mode.

Reply:
I think the only confusion on my side is that setting `noSNAT` to true in `networkPolicyOnly` mode will give an error, even though in practice we don't do any SNAT in that case. But I think a comment to that effect in the `noSNAT` config parameter description is good enough.

Reply:
Added comments and changed to ignore the option for networkPolicyOnly mode.